nengo_gui/components/bg_plot.py

```diff
@@ -0,0 +1,52 @@
+import struct
+
+import nengo
+import numpy as np
+
+from nengo_gui.components.value import Value
+
+
+class BGPlot(Value):
+    """The server-side system for the SPA Basal Ganglia plot."""
+
+    # the parameters to be stored in the .cfg file
+    config_defaults = Value.config_defaults
+    config_defaults["palette_index"] = 1
+    config_defaults["show_legend"] = True
+
+    def __init__(self, obj, **kwargs):
+        args = kwargs["args"]
+        super(BGPlot, self).__init__(obj, args["n_lines"])
+
+        # default legends to show
+        self.def_legend_labels = args["legend_labels"]
+
+        # the item to connect to
+        self.probe_target = args["probe_target"]
+
+        self.label = "bg " + self.probe_target
+
+    def attach(self, page, config, uid):
+        super(Value, self).attach(page, config, uid)
+
+    def add_nengo_objects(self, page):
+        # create a Node and a Connection so the Node will be given the
+        # data we want to show while the model is running.
+        with page.model:
+            self.node = nengo.Node(self.gather_data,
+                                   size_in=self.n_lines)
+            if self.probe_target == "input":
+                self.conn = nengo.Connection(self.obj.input, self.node, synapse=0.01)
+            else:
+                self.conn = nengo.Connection(self.obj.output, self.node, synapse=0.01)
+
+    def javascript(self):
+        # generate the javascript that will create the client-side object
+        info = dict(uid=id(self), label=self.label, 
+                    n_lines=self.n_lines, synapse=0)
+
+        if getattr(self.config, "legend_labels") == []:
+            json = self.javascript_config(info, override={"legend_labels":self.def_legend_labels})
```

Comment:
Because the behaviour of `javascript_config` is to override everything with the defaults. It doesn't look to see if the defaults have been set to anything else. Is that it's intended behaviour or should the code be changed?